libminifi/src/processors/ExtractText.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file ExtractText.cpp
+ * ExtractText class implementation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <iterator>
+#include <string>
+#include <memory>
+#include <set>
+
+#include "processors/ExtractText.h"
+#include "core/ProcessContext.h"
+#include "core/ProcessSession.h"
+#include "core/FlowFile.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+core::Property ExtractText::Attribute("Attribute", "Attribute to set from content (TEMPORARY)", "");
```

Comment:
what does TEMPORARY mean in this context?

Reply:
Not sure, ask @achristianson. I got it from [here](https://github.com/NiFiLocal/nifi-minifi-cpp/commit/ec03fd15f80e33f1dba3be9c1e2520745e732930#diff-3d4ecb8ad9604bc01f1c18e283d8bc4fR31).

Reply:
@calebj @phrocker In the [reference NiFi ExtractText](https://nifi.apache.org/docs/nifi-docs/components/org.apache.nifi/nifi-standard-nar/1.4.0/org.apache.nifi.processors.standard.ExtractText/index.html), the attribute to set from the extracted text, as well as the regular expression to match, is set with dynamic properties. Until MiNiFi - C++ supports dynamic properties, the intent was for ExtractText to support one regular expression and one attribute. The "TEMPORARY," if I recall correctly, was to document that this property is liable to be deprecated in the future. Looking back at it, we should probably clarify this, or remove the text altogether. Also suggest we prioritize the dynamic attributes feature.